### PR TITLE
Implement Discord Client and Command Handler

### PR DESCRIPTION
**Pull request by Charly6596**
## Related issue
#1

## Changes
Implemented a basic client connection, using DSharpPlus.
- With a `CommandHandler`, that have a method, `InitializeAsync()`, to setup the DSharpPlus `CommandsNextModule`. This setup involves actions as registering the commands, or converters.
- With a `DSharpClient`, that have a method `ConnectAsync()`, wich basically, connects the client.
- And `MiunieDiscord`, that is the connection between Miunie.Core and Miunie.Discord. 

# Expected Feedback
I wanted to modify the name of `CommandHandler`, since it's not handling any command, in fact, DSharpPlus already got a command handler implemented, but i couldn't figure out a name for that.

Regarding to the issue #2 I don't see a way the `BotConfiguration` would fit as a dependency of `DSharpClient`, pheraps I missunderstood the concept? We'd need a class, kind of a factory, to generate the configuration. That class would depend on `BotConfiguration`.

Any kind of constructive feedback is always welcome :)

### CODE REVIEW

File: src/Miunie.Discord/DSharpClient.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        private DiscordClient _discordClient;
+
+        public DSharpClient(DiscordClient discordClient)
+        {
+            _discordClient = discordClient;
+        }
+
+        public async Task ConnectAsync()
+        {
+            await _discordClient.ConnectAsync();
```
Comment (petrspelos) [inline src/Miunie.Discord/DSharpClient.cs]: This class is redundant. Let's get rid of it. The `DSharpClient` in the Architecture drawing was just to illustrate that we will use the Client from DSharp.

File: src/Miunie.Discord/CommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            _commandsNextModule = commandsNextModule;
+        }
+
+        public async Task InitializeAsync()
+        {
+            /*
+             TODO (Charly): Here we should register the commands, converters...
+             using the _commandService.RegisterCommands method
+             https://dsharpplus.emzi0767.com/api/DSharpPlus.CommandsNext.CommandsNextModule.html#DSharpPlus_CommandsNext_CommandsNextModule_RegisterCommands_Assembly_
+            */
```
Comment (petrspelos) [inline src/Miunie.Discord/CommandHandler.cs]: This also feels like an empty proxy and I think we should get rid of this class.
The `CommandHandler` was meant as the command handler system that comes with DSharpPlus.

If you say that only `CommandsNextModule` is needed, we should just have that directly in `MiunieDiscord`

**Comment by petrspelos** [comment]
@Charly6596 as for your question about #2:
With the changes that I just requested, the `BotConfiguration` would be injected into `DiscordMiunie`.

The `BotConfiguration` will be a class that will hold a `GetDiscordBotToken` method.

`DiscordMiunie` will use `BotConfiguration` to get the token in order to construct the Configuration needed for a client. That part is not implemented in your version of the code:

https://github.com/DSharpPlus/Example-Bots/blob/master/DSPlus.Examples.CSharp.Ex03/Program.cs#L63&L71

**Comment by Charly6596** [comment]
@petrspelos That makes a lot of sense, indeed it is redundant, I missunderstood the diagram, will fix it asap 👍

File: src/Miunie.Discord/Configuration/IBotConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Miunie.Discord.Configuration
+{
+    public interface IBotConfiguration
+    {
+        string GetBotToken();
+    }
+}
```
Comment (petrspelos) [inline src/Miunie.Discord/Configuration/IBotConfiguration.cs]: Whops, missed a newline here.

You can actually set it to add a newline at the end of a file in VS.

File: src/Miunie.Discord/MiunieDiscord.cs [hunk 4]
```diff
@@ -1,23 +1,67 @@
...
         public async Task RunAsync()
         {
-            await _client.ConnectAsync();
-            await _commandHandler.InitializeAsync();
+
+            await InitializeDiscordClientAsync();
+
+            await InitializeCommandsNextModuleAsync();
+
+        }
```
Comment (petrspelos) [inline src/Miunie.Discord/MiunieDiscord.cs]: Can we get rid of the empty line 22 and 27? Just so it's more consistent with the rest of our code? :slightly_smiling_face:

**Pull request merged**